### PR TITLE
Enable SageTV to load plugins from a SageTVPluginsDev.d directory

### PR DESCRIPTION
**Pull request by stuckless**
See the SageTVPluginsDev.md README for information about this change.  If there is still questions, I'll answer them :)

### CODE REVIEW

File: java/sage/FileDownloader.java [hunk 1]
```diff
@@ -152,7 +153,7 @@ public synchronized Object downloadFile(String serverName, String srcFile, java.
     if (Sage.client && serverName == null)
       serverName = Sage.preferredServer;
     remoteUIXfer = uiMgr != null && uiMgr.getUIClientType() == UIClient.REMOTE_UI && uiMgr.hasRemoteFSSupport();
-    if (serverName != null && (serverName.startsWith("http:") || serverName.startsWith("https:") || serverName.startsWith("ftp:")))
+    if (serverName != null && (serverName.startsWith("http:") || serverName.startsWith("https:") || serverName.startsWith("ftp:") ||serverName.startsWith("file:")))
```
Comment (Narflex) [inline java/sage/FileDownloader.java]: Put a space before 'serverName.startsWith("file:")

File: java/sage/FileDownloader.java [hunk 2]
```diff
@@ -376,6 +391,10 @@ else if (remoteUIXfer)
       {
         remoteUITaskRun();
       }
+      if (myServerName!=null && myServerName.startsWith("file:"))
```
Comment (Narflex) [inline java/sage/FileDownloader.java]: This should be an 'else if'

File: java/sage/FileDownloader.java [hunk 3]
```diff
@@ -396,6 +415,27 @@ else if (remoteUIXfer)
     }
   }
 
+  private void fileTaskRun() {
+    try {
+      downloadedBytes=myDestFile.length();
```
Comment (Narflex) [inline java/sage/FileDownloader.java]: Clearly this is cheating since it's not reporting real-time progress...but I'm fine with it. :)

**Comment by Narflex** [comment]
Just fix the nit and the one error in the code and then I'll merge.

**Comment by stuckless** [comment]
Thanks for the review... I added the changes.

**Pull request merged**